docs/pages/setup/reference/cli.mdx

```diff
@@ -743,6 +745,46 @@ add_headers:
       "Authorization": "Bearer tokentokentoken"
 ```
 
+#### Aliases
+
+You can define custom `tsh` command aliases using the `aliases` section. It also allows for changing of the default flag values for existing commands.
+
+New command `tsh l`:
+
+```yaml
+aliases:
+    "l": "tsh login --auth=okta --username=alice"
+```
+
+Make `tsh status` use JSON as a default format:
+
+```yaml
+aliases:
+    "status": "tsh status --format=json"
+```
+
+Define custom command using with `bash`. The `$0` and `$1` variables will be substituted by command arguments.
+
+```yaml
+aliases:
+    "connect": "bash -c 'tsh login $0 && tsh ssh $1'"
+```
+
+The alias can use an arbitrary number of arguments. If a argument variable `$N` is referenced, `tsh` will check that at least `N+1` arguments were given to the alias invocation. All arguments that were given but not referenced in the alias definition will be appended at the end.
+
+Given configuration:
+
+```yaml
+aliases:
+    "example": "bash -c 'echo first=$0 $0-$1 $3'"
```

Comment:
Interesting that we can allow for executing non-tsh commands. Did we do a security review of this feature? I'd be concerned about somebody dropping a config file in a particular location to create an alias that invokes some malicious code.

Reply:
It's the same as with e.g. git aliases which allow you to escape to a shell (only with a bit different syntax) - aliases are local to the user's system and defined in tsh config so if an attacker has access to the system I think they can do more damage than override your aliases. I agree a security review is a good idea though.

Reply:
I can imagine an attack where the `--proxy` value is changed to an attacker-controlled modified Teleport cluster which captures any credentials used for logging in.

However, attacker capable of writing `~/.tsh/config/config.yaml` or `/etc/tsh.yaml` is likely to be able to modify `~/.bashrc` or equivalent. This is probably an easier and more stable exploit, too.